website_sale_one_step_checkout_delivery/controllers/main.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# © 2017 bloopark systems (<http://bloopark.de>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import http
+from odoo.addons.website_sale_one_step_checkout.controllers.main import \
+    WebsiteSale
+from odoo.http import request
+
+
+class WebsiteSaleOneStepCheckoutDelivery(WebsiteSale):
+
+    @http.route(['/shop/checkout/change_delivery'], type='json', auth="public",
+                website=True, multilang=True)
+    def change_delivery(self, **post):
+        """If delivery method was changed in frontend.
+
+        Change and apply delivery carrier / amount to sale order.
+        """
+        order = request.website.sale_get_order()
+        carrier_id = int(post.get('carrier_id'))
```

Comment:
no need to do `get()` if it always will have carrier_id, otherwise it can fail when `get()` will return `False`

Reply:
"otherwise it can fail when `get()` will return False". Could elaborate on this a bit, please? I'm afraid I can not follow.

Reply:
[get()](https://docs.python.org/2.7/library/stdtypes.html#dict.get) returns `False` if there is no key in it, so int(False) will fail.

Reply:
`int(False)` should be actually broadcasted to `0`. But since `get` defaults to `None` and `None` can not be broadcasted, this will break it. So you are right. Thanks for pointing it out. I will fix this.

Reply:
Sorry, yeah, it's `None`.